.travis.yml

```diff
@@ -26,3 +47,5 @@ script:
   - make
   - if [ $USE_CMAKE == "YES" ] ; then CTEST_OUTPUT_ON_FAILURE=1 make test ; fi
   - if [ $USE_CMAKE == "NO" ] ; then make check ; fi
+  - cppcheck --language=c --std=c89 --verbose --quiet --error-exitcode=1 --enable=warning,performance,portability src/
```

Comment:
I just tested on a computer on Debian 9 (current testing). `--language=c` is optional, so it could be removed.
Travis CI works only with GitHub. So I don't personally use it, it just seems simple to add a new check, so I did it. GitHub is centralized, based on proprietary software, send proprietary JavaScript, so I only use it for third party projects, and even wants its death. http://blog.printf.net/articles/2015/05/29/announcing-gittorrent-a-decentralized-github/ https://www.gnu.org/philosophy/who-does-that-server-really-serve.html https://carlchenet.com/2016/01/22/le-danger-github/
You are right cppcheck analyze the source code, it is platform and compilation independent.
I never use Jenkins, so I can help you with it.